File: backend/firebase/database/entidades/Cupon.py
class Cupon:
    def __init__(self, id = 0, nombre = "", descripcion = ""):
        self._id = id
        self._nombre = nombre
        self._descripcion = descripcion
        self._imagen = imagen

    @property
    def id(self):
        return self._id

    @id.setter
    def id(self, id):
        self._id = id

    @property
    def nombre(self):
        return self._nombre
    
    @nombre.setter
    def nombre(self, nombre):
        self._nombre = nombre

    @property
    def descripcion(self):
        return self._descripcion
    
    @descripcion.setter
    def descripcion(self, descripcion):
        self._descripcion = descripcion

    @property
    def imagen(self):
        return self._imagen
    
    @imagen.setter
    def imagen(self, imagen):
        self._imagen = imagen

    def toString(self):
        return f"Cupon {self.id}: {self.nombre}, {self.descripcion}, {self.imagen}"

File: frontend/src/componentes/Cupon/Cupon.js
import React, { useEffect, useState } from "react";
import style from "./Cupon.module.css"
import * as CuponServer from './CuponServer';
import { recuperarBusqueda } from "../NavBar/NavBar";
import ModalCupon from "./ModalCupon";


let [cupones, setCupones] = [];
var conexionBD = false;
const Cupon = () => {
    [cupones, setCupones] = useState([]);

    useEffect(() => {
        listaCupones(null);
    },[]);

    function comprobacion(conexionBD){
        if(conexionBD === false){
            return <ModalCupon/>
        }else{
            return (<><div id={style.cabecera}>
                
            </div>
    
            <div id={style.contenedorTarjetas}>
                {cupones.map(cupon => (
                    <div id={style.tarjeta} className="card" key={cupon.id}>
                        <img id={style.imgCard} src={cupon.imagen} className="card-image-top" alt=""/>
                        <div className="card-body" id={style.cuerpoTarjeta}>
                            <h5>{cupon.nombre}</h5>
                            <p>{cupon.descripcion}</p>
                        </div>
                    </div>
                ))}
            </div></>)
        }
    }
    return(
        comprobacion(conexionBD)
    );
};

export default Cupon;

export const listaCupones = async (busqueda) => {
    try{
        const data = await (await CuponServer.getCupones()).json();
        if(busqueda == null) setCupones(data.Cupones);
        else setCupones(recuperarBusqueda(busqueda, data.Cupones));
        conexionBD = true;
    }catch(error){
        console.log(conexionBD);
        console.log(error);
    }
};